sn_node/src/replication.rs

```diff
@@ -57,10 +57,12 @@ impl Node {
         }
 
         // set the distance range used for pruning
-        let distance_range = match our_close_group.get(CLOSE_GROUP_SIZE) {
+        // our_close_group now only have CLOSE_GROUP_SIZE entries,
+        // which makes `our_close_group.get(CLOSE_GROUP_SIZE)` return None always.
```

Comment:
I'm not sure I follow why this is always None?

Reply:
`our_close_group` now only contains 8th entries, 
but `.get(CLOSE_GROUP_SIZE)` is asking for the entry with `index 8`.
As the vec's indexing is from 0, hence it's actually looking from `9th entry`, which is `None`

Reply:
Ah, that comment is mainly explain what's the issue of previous, i.e. `our_close_group.get(CLOSE_GROUP_SIZE)` .
Not about the current approach (using `last()`).